### PR TITLE
Kill off duplication and make Block/Item classes so much neater

### PR DESCRIPTION
**Pull request by dktapps**
### Description
Clean up Block and Item classes and reduce duplication using interfaces containing the relevant block/item IDs. The access methods of these constants remains exactly the same as it did before.

### Reasons to modify
- Less mess
- Better consistency
- Less duplication

### Tests & Reviews
I have tested this and these changes do not affect or change functionality at all; all original functionality is retained. There are no compatibility issues with older APIs.

### CODE REVIEW

**Comment by dschwartz783** [comment]
Is there any reason you'd want to implement BlockId or ItemId in any other class though?

File: src/pocketmine/item/ItemIds.php [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+	const BEETROOT_SEEDS = 458; const BEETROOT_SEED = 458;
+	const BEETROOT_SOUP = 459;
+	const RAW_SALMON = 460;
+	const CLOWN_FISH = 461;
+	const PUFFER_FISH = 462;
+	const COOKED_SALMON = 463;
+
+	const ENCHANTED_GOLDEN_APPLE = 466;
+
+	const CAMERA = 498; #blamemojang
```
Comment (JUSTaMod) [inline src/pocketmine/item/ItemIds.php]: lol. blame mojang is right.

**Comment by ishitatsuyuki** [comment]
I will leave you to choose: but expressing my opinion, Blocks shouldn't be mixed with Items. There are items not a block, blocks not an item. Items and Blocks are essentially different thing, just having many common IDs.

**Comment by dschwartz783** [comment]
The logical separation is important to avoid confusion, Imo.

**Comment by ZedCee** [comment]
This should improve code efficiency and maybe even increase speed a little by limiting the size of called namespaces for any given function, shouldn't it?

_PHP is a guessing game for me, but I would assume it decompresses the phar into a virtual environment of sorts.  If the phar is decompressed on-the-fly, then any improvement would only be evident running from src/..._

**Comment by ZedCee** [comment]
@dktapps Haven't had enough time to really see and I couldn't imagine how to benchmark this: Start-up seems super snappy and I don't segfault on start anymore.
Stopping is another story, as it seems a segfault IS how it quits...   Wait, where are my core.:hankey:s?

Shit. 
After the 'rm -rf' incident (Damned jenkins.sh), I forgot to turn those off :worried: 
Oh dear...  /tmp...  
Gotta go

**Comment by ishitatsuyuki** [comment]
No, this won't have a dramatic improvement. @ZedCee Please stop spamming with your noob knowledges.

**Comment by dktapps** [comment]
@ishitatsuyuki that's why I split them up into two separate classes. I've made all Block constants accessible as Item constants also for the purposes of backwards compatibility.

**Comment by ishitatsuyuki** [comment]
Well, if it retains the compatibility, this should be OK since it's the Mojang's design and it's just the ID part. Code review +1

**Comment by PeratX** [comment]
+1

**Comment by dktapps** [comment]
@ishitatsuyuki ...?

**Comment by dktapps** [comment]
I'm done with that branch now

**Pull request merged**